docs/reference/migration/apis/data-stream-reindex.asciidoc

```diff
@@ -0,0 +1,358 @@
+
+[[reindex-data-stream-api-settings]]
+==== Settings
+
+You can use the following settings to control the behavior of the reindex data stream API:
+
+[[migrate_max_concurrent_indices_reindexed_per_data_stream-setting]]
+// tag::migrate_max_concurrent_indices_reindexed_per_data_stream-setting-tag[]
+`migrate.max_concurrent_indices_reindexed_per_data_stream`
+(<<dynamic-cluster-setting,Dynamic>>)
+The number of backing indices within a given data stream which will be reindexed concurrently. Defaults to `1`.
+// end::migrate_max_concurrent_indices_reindexed_per_data_stream-tag[]
+
+[[migrate_data_stream_reindex_max_request_per_second-setting]]
+// tag::migrate_data_stream_reindex_max_request_per_second-setting-tag[]
+`migrate.data_stream_reindex_max_request_per_second`
+(<<dynamic-cluster-setting,Dynamic>>)
+The average number of documents within a given backing index to reindex per second.
+Defaults to `1000`, though can be any decimal number greater than `0`.
+To remove throttling, set to `-1`.
+This setting can be used to throttle the reindex process and manage resource usage.
+Consult the <<docs-reindex-throttle,reindex throttle docs>> for more information.
+// end::migrate_data_stream_reindex_max_request_per_second-tag[]
+
+
+[[reindex-data-stream-api-example]]
+==== {api-examples-title}
+
+Assume we have a data stream `my-data-stream` with the following backing indices, all of which have index major version 7.x.
+
+* .ds-my-data-stream-2025.01.23-000001
+* .ds-my-data-stream-2025.01.23-000002
+* .ds-my-data-stream-2025.01.23-000003
+
+Let's also assume that `.ds-my-data-stream-2025.01.23-000003` is the write index.
+If {es} is version 8.x and we wish to upgrade to major version 9.x, the version 7.x indices must be upgraded in preparation.
+We can use this API to reindex a data stream with version 7.x backing indices and make them version 8 backing indices.
+
+Start by calling the API:
+
+[[reindex-data-stream-start]]
+[source,console]
+----
+POST _migration/reindex
+{
+    "source": {
+        "index": "my-data-stream"
+    },
+    "mode": "upgrade"
+}
+----
+// TEST[skip:cannot easily clean up reindex task between tests]
+
+
+As this task runs in the background this API will return immediately.
+The task will do the following.
+
+First, the data stream is rolled over. So that no documents are lost during the reindex, we add <<index-block-settings,write blocks>>
+ to the existing backing indices before reindexing them. Since a data stream's write index cannot have a write block,
+ the data stream is must be rolled over. This will produce a new write index, `.ds-my-data-stream-2025.01.23-000004`; which
+  has an 8.x version and thus does not need to be upgraded.
+
+Once the data stream has a write index with an 8.x version we can proceed with reindexing the old indices.
+For each of the version 7.x indices, we now do the following:
+
+* Add a write block to the index to guarantee that no writes are lost.
```

Comment:
Should we include open and re-close closed indices here? Or too low-level of a detail?